executor/insert_common.go

```diff
@@ -215,34 +218,36 @@ func (e *InsertValues) getRow(cols []*table.Column, list []expression.Expression
 		row[offset], hasValue[offset] = val1, true
 	}
 
-	return e.fillGenColData(cols, len(list), hasValue, row)
+	return e.fillRow(row, hasValue)
 }
 
-// fillDefaultValues fills a row followed by these rules:
+// setValueForRefColumn set some default values for the row to eval the row value with other columns,
+// it follows these rules:
 //     1. for nullable and no default value column, use NULL.
 //     2. for nullable and have default value column, use it's default value.
 //     3. for not null column, use zero value even in strict mode.
 //     4. for auto_increment column, use zero value.
 //     5. for generated column, use NULL.
-func (e *InsertValues) fillDefaultValues(row []types.Datum, hasValue []bool) error {
+func (e *InsertValues) setValueForRefColumn(row []types.Datum, hasValue []bool) error {
 	for i, c := range e.Table.Cols() {
-		var err error
-		if c.IsGenerated() {
-			continue
```

Comment:
Is this behavier has be changed~? maybe need a test for  ref column generate insert ?

Reply:
I think we have a case for it. Let me check.

Reply:
``getColDefaultValue` will return null for generated columns.